Added a method rowAdd(...) to Matrix

christianbender opened the pull request:
Issue #64 
I added the feature as requested:  

> Adds a multiple of one row to another inside a matrix.

```js
rowAdd(dest : number, source : number, scalar : number = 1.0 ) : Matrix
```

Inclusive tests

mateogianolio commented:
Thanks, will have a look at it this weekend.

christianbender commented:
@christianbender Super, Thanks.

mateogianolio commented on src/Matrix.prototype.spec.ts:
```diff
@@ -343,4 +343,19 @@ describe('Matrix.prototype', () => {
+      let m = new Matrix([[1, 2], [3, 4]]);
+      assert.throws(m.rowAdd.bind(3, 1, 10), Error);
+      assert.throws(m.rowAdd.bind(0, 5, 10), Error);
+      assert.throws(m.rowAdd.bind(0, 1, 'a'), Error);
```
This line causes the build to fail. Remove it.

mateogianolio commented on src/Matrix.ts:
```diff
@@ -801,6 +801,41 @@ export default class Matrix extends NDArray {
+    // preconditions
+    this.check(dest, 0);
+    this.check(source, 0);
+    if (typeof scalar != "number")
```
There are no runtime type checks in the other methods, no reason to do it here.

mateogianolio replied:
@christianbender Sorry, I only meant the `if (typeof scalar != "number")`! The out of bounds checks (`this.check(dest, 0)` and `this.check(source, 0)`) should definitely still be there :)

christianbender commented:
@mateogianolio 
* I changed the method ```rowAdd(...)``` as requested and made it more effiecient. 
* In addition I removed the runtime type checks and changed the test-suite accordingly. Removed the exception-test

christianbender commented:
@mateogianolio Done. I added the preconditions again
